raptorWeb/authprofiles/views.py

```diff
@@ -0,0 +1,347 @@
+
+class RegisterUser(TemplateView):
+
+    template_name = join(settings.AUTH_TEMPLATE_DIR, 'registration.html')
+    registered = False
+    register_form = UserForm()
+    extra_form = UserProfileInfoForm()
+
+    def get(self, request):
+
+        dictionary = player_poller.currentPlayers_DB
+        dictionary["registered"] = self.registered
+        dictionary["register_form"] = self.register_form
+        dictionary["extra_form"] = self.extra_form
+        dictionary = viewContext.update_context(context = dictionary)
+        
+        return render(request, self.template_name, context=dictionary)
+
+    def post(self,request):
+
+        register_form = UserForm(request.POST)
+        extra_form = UserProfileInfoForm(request.POST)
+
+        dictionary = player_poller.currentPlayers_DB
+        dictionary["registered"] = self.registered
+        dictionary["register_form"] = register_form
+        dictionary["extra_form"] = extra_form
+        dictionary = viewContext.update_context(context = dictionary)
+
+        if register_form.is_valid() and extra_form.is_valid():
+
+            LOGGER.info("A new User has been registered!")
+            new_user = register_form.save()
+            new_user.set_password(new_user.password)
+            new_user.save()
+            new_user_extra = extra_form.save(commit=False)
+            new_user_extra.user = new_user
+            if "profile_picture" in request.FILES:
+                new_user_extra.profile_picture = request.FILES["profile_picture"]
+            new_user_extra.save()
+            registered = True
+            dictionary["registered"] = registered
+
+            return render(request, self.template_name, context=dictionary)
+
+        else:
+
+            dictionary["register_form"] = register_form
+            dictionary["extra_form"] = extra_form
+
+            return render(request, self.template_name, context=dictionary)
+
+class User_Login_Form(TemplateView):
+    """
+    Returns a form for a user to login with Username and Password
+    """
+    login_form = UserLoginForm()
+
+    def get(self, request):
+
+        if request.headers.get('HX-Request') == "true":
+            dictionary = player_poller.currentPlayers_DB
+            dictionary["login_form"] = self.login_form
+            template_name = join(settings.AUTH_TEMPLATE_DIR, 'login.html')
+            return render(request, template_name, context=dictionary)
+        else:
+            return HttpResponseRedirect('../')
+
+    def post(self, request):
+
+        login_form = UserLoginForm(request.POST)
+        dictionary = player_poller.currentPlayers_DB
+        dictionary["login_form"] = self.login_form
+
+        if login_form.is_valid():
+            username = login_form.cleaned_data["username"]
+            password = login_form.cleaned_data["password"]
+            user = authenticate(username=username, password=password)
+            if user:
+                LOGGER.info("User logged in!")
```

Comment:
More verbose logging about the logged-in user would be better than a simple statement.